recipes/ssht/all/conanfile.py

```diff
@@ -14,53 +14,51 @@ class SshtConan(ConanFile):
     default_options = {"fPIC": True}
     requires = "fftw/3.3.9"
     generators = "cmake"
-    exports_sources = [
-        "src/c/*",
-        "include/ssht/*.h",
-        "tests/CMakeLists.txt",
-        "tests/*.c",
-        "tests/*.h",
-        "CMakeLists.txt",
-        "cmake/*.cmake",
-    ]
+    exports_sources = ["CMakeLists.txt"]
 
     @property
-    def _src_dir(self):
-        return f"{self.name}-{self.version}"
+    def _source_subfolder(self):
+        return "source_subfolder"
 
-    def configure(self):
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
         if self.settings.compiler == "Visual Studio":
             raise ConanInvalidConfiguration("SSHT requires C99 support for complex numbers.")
         del self.settings.compiler.cppstd
         del self.settings.compiler.libcxx
 
+    def source(self):
+        from os import rename, getcwd
+        from shutil import copytree, rmtree
+        from glob import glob
+
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob('ssht-*/')[0]
+        rename(extracted_dir, self._source_subfolder)
+
     @property
     def cmake(self):
         if not hasattr(self, "_cmake"):
             self._cmake = CMake(self)
-            self._cmake.definitions["tests"] = True
-            self._cmake.definitions["conan_deps"] = True
+            self._cmake.definitions["tests"] = False
             self._cmake.definitions["python"] = False
-            self._cmake.definitions["CONAN_CENTER"] = True
-            self._cmake.configure(build_folder="build", source_folder=self._src_dir)
+            self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
     def build(self):
-        from pathlib import Path
-
-        path = Path(self.source_folder)
-        build = Path(self.source_folder) / "build"
-        build.mkdir(exist_ok=True)
-        (path / "conanbuildinfo.cmake").rename(path / "build" / "conanbuildinfo.cmake")
         self.cmake.build()
 
     def package(self):
-        self.copy("LICENSE", dst="licenses", src=self._src_dir)
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
         self.cmake.install()
 
     def package_info(self):
-        self.cpp_info.libs = ["ssht"]
-        self.cpp_info.builddirs = ["lib/cmake/ssht"]
+        from os.path import join
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
+        self.cpp_info.names["cmake_find_package"] = "SSHT"
+        self.cpp_info.names["cmake_find_package_multi"] = "SSHT"
```

Comment:
There's no namespace upstream https://github.com/astro-informatics/ssht/blob/061763ac1140fb77b16301ef45fe1069b1f2795c/src/c/CMakeLists.txt#L19
This is not required to be implemented https://github.com/conan-io/conan-center-index/blob/770d232151108426f4fadf48e2967434865c96a0/recipes/tesseract/all/conanfile.py#L131 but please add a FIXME so we know


This does not match the `FindPackage(<NAME>)` https://github.com/astro-informatics/ssht/blob/061763ac1140fb77b16301ef45fe1069b1f2795c/cmake/exporting.cmake#L12
the `cpp_info` has a `filenames` for that (same syntax as names)